File: armario/generar.py
import os
import json

# Carpeta raíz de assets
assets_root = "assets"

# Extensiones válidas
valid_ext = (".png", ".jpg", ".jpeg", ".avif")

# Buscar todas las subcarpetas dentro de assets
folders = [
    os.path.join(assets_root, name)
    for name in os.listdir(assets_root)
    if os.path.isdir(os.path.join(assets_root, name))
]

# Procesar cada carpeta
for folder in folders:
    files = [
        f"{folder}/{f}" for f in os.listdir(folder)
        if f.lower().endswith(valid_ext)
    ]
    if files:
        json_path = os.path.join(folder, "files.json")
        with open(json_path, "w") as f:
            json.dump(files, f, indent=2)
        print(f"✅ {len(files)} imágenes encontradas en {folder}, archivo {json_path} creado")
    else:
        print(f"⚠️ No se encontraron imágenes en {folder}")


#python3 -m http.server 8000

File: armario/generar.java
import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

public class Generar {
    public static void main(String[] args) {
        File assetsRoot = new File("assets");

        if (!assetsRoot.exists() || !assetsRoot.isDirectory()) {
            System.out.println("❌ No se encontró la carpeta 'assets'");
            return;
        }

        File[] subfolders = assetsRoot.listFiles(File::isDirectory);

        if (subfolders != null) {
            for (File folder : subfolders) {
                List<String> validFiles = new ArrayList<>();
                File[] files = folder.listFiles();

                if (files != null) {
                    for (File f : files) {
                        String name = f.getName().toLowerCase();
                        if (name.endsWith(".png") || name.endsWith(".jpg") || 
                            name.endsWith(".jpeg") || name.endsWith(".avif")) {
                            // Guardamos la ruta relativa exacta que espera tu HTML
                            validFiles.add("assets/" + folder.getName() + "/" + f.getName());
                        }
                    }
                }

                if (!validFiles.isEmpty()) {
                    saveJson(new File(folder, "files.json"), validFiles);
                    System.out.println("✅ Generado files.json en: " + folder.getName());
                }
            }
        }
    }

    private static void saveJson(File outFile, List<String> data) {
        try (FileWriter writer = new FileWriter(outFile)) {
            writer.write("[\n");
            for (int i = 0; i < data.size(); i++) {
                writer.write("  \"" + data.get(i) + "\"" + (i < data.size() - 1 ? "," : "") + "\n");
            }
            writer.write("]");
        } catch (IOException e) {
            System.out.println("❌ Error al escribir JSON: " + e.getMessage());
        }
    }
}
